packages/@ourworldindata/explorer/src/Explorer.tsx

```diff
@@ -686,128 +666,111 @@ export class Explorer
+        // together. The order is arbitrary, but should be left as is for backward compatibility.
+        if (config.ySlugs && explorerGrapherConfig.yVariableIds)
+            config.ySlugs = `${config.ySlugs} ${explorerGrapherConfig.yVariableIds}`
 
         this.inputTableTransformer = (table: OwidTable) => {
-            // add transformed (and intermediate) columns to the grapher table
-            if (uniqueSlugsInGrapherRow.length) {
-                const allColumnSlugs = _.uniq(
-                    uniqueSlugsInGrapherRow.flatMap((slug) => [
-                        ...this.getBaseColumnsForColumnWithTransform(slug),
-                        slug,
-                    ])
+            // All slugs specified by the author in the explorer config
+            const slugs = _.uniq(slugDimensions.map(({ slug }) => slug))
+
+            // Add transformed columns (and the variables they depend on) to the grapher table
+            if (slugs.length > 0) {
+                const baseSlugs = slugs.flatMap((slug) =>
+                    this.getDataSlugsForTransformedColumn(slug)
                 )
-                const existingColumnSlugs = table.columnSlugs
-                const outstandingColumnSlugs = allColumnSlugs.filter(
-                    (slug) => !existingColumnSlugs.includes(slug)
+                const allRequiredSlugs = _.uniq([...baseSlugs, ...slugs])
+                const existingSlugs = new Set(table.columnSlugs)
+                const missingSlugs = allRequiredSlugs.filter(
+                    (slug) => !existingSlugs.has(slug)
                 )
-                const requiredColumnDefs = outstandingColumnSlugs
-                    .map(
-                        (slug) =>
-                            this.columnDefsWithoutTableSlugByIdOrSlug[slug]
-                    )
+                const missingColumnDefs = missingSlugs
+                    .map((slug) => columnDefsWithoutTableSlugByIdOrSlug[slug])
                     .filter(_.identity)
-                table = table.appendColumns(requiredColumnDefs)
+                table = table.appendColumns(missingColumnDefs)
             }
 
-            // update column definitions with manually provided properties
-            table = table.updateDefs((def: OwidColumnDef) => {
-                const manuallyProvidedDef =
-                    this.columnDefsWithoutTableSlugByIdOrSlug[def.slug] ?? {}
-                const mergedDef = { ...def, ...manuallyProvidedDef }
-
-                // update display properties
-                mergedDef.display ??= {}
-                if (manuallyProvidedDef.name)
-                    mergedDef.display.name = manuallyProvidedDef.name
-                if (manuallyProvidedDef.unit)
-                    mergedDef.display.unit = manuallyProvidedDef.unit
-                if (manuallyProvidedDef.shortUnit)
-                    mergedDef.display.shortUnit = manuallyProvidedDef.shortUnit
-
-                return mergedDef
-            })
```

Comment:
_I think_ this can be safely dropped now that the display properties are marked as such and added to the display object when parsed